--- /src/gyp/common/gypi
+++ /src/gyp/common.gypi
@@ -213,6 +207,10 @@
           ['compiler_target=="clang" or compiler_target=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
+            ],
+          }],
+          ['compiler_target=="clang"', {
+            'cflags_cc': [
               '-stdlib=libc++',
             ],
           }],
@@ -233,6 +231,10 @@
           ['compiler_host=="clang" or compiler_host=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
+            ],
+          }],
+          ['compiler_target=="clang"', {
+            'cflags_cc': [
               '-stdlib=libc++',
             ],
           }],
